internal/providers/applehv/applehv.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2023 Red Hat
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package applehv
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"os/exec"
+
+	"github.com/coreos/ignition/v2/config/v3_5_experimental/types"
+	"github.com/coreos/ignition/v2/internal/distro"
+	"github.com/coreos/ignition/v2/internal/platform"
+	"github.com/coreos/ignition/v2/internal/providers/util"
+	"github.com/coreos/ignition/v2/internal/resource"
+	"github.com/coreos/vcontext/report"
+	"github.com/mdlayher/vsock"
+)
+
+/*
+  To test this with qemu, we should simply add a vsock connection to the qemu call.  We would then
+  need an httpd server that is capable of responding to a GET over vsock.  Alternatively, the httpd
+  server could listen on a unix domain socket (uds) and a utility would need to be written that copies
+  from the vsock <-> uds
+*/
```

Comment:
Is this a correct "diagram" of the platform setup?

HTTP server <--> vsock (host) <--> QEMU <--> kernel (vsock-virtio) <--> Ignition (vsock client)